src/project_model.rs

```diff
@@ -28,8 +29,7 @@ pub struct Package(usize);
 
 #[derive(Debug)]
 struct PackageData {
-    manifest: PathBuf,
-    lib: Option<PathBuf>,
+    lib: Option<(PathBuf, String)>,
```

Comment:
This `String` doesn't feel exactly right: package itself does not really know its library name: the same pacakge might be used with different names by different dependencies:

```
# one crate
[dependecies]
bar = { package = "foo"}

# another
[dependecies]
baz = { package = "foo"}
```

Here, `foo`'s library is known under the name of `bar` or `baz`, depending on which package are you asking. 
If I understand correctly, we use this mainly for self-dependneices in tests. Perhaps we should adjust Cargo to always include self-dependency? Will try to look into that, but, until than, current solution is actually fine!

Reply:
OMG, I didn't know cargo is such a complex tool :astonished: